### PR TITLE
fix(DateRangeInputs): BED-6514 resolve date range validation edge case

### PR DESCRIPTION
**Pull request by TheNando**
## Description

* Improved DateRangeInputs validation logic

## Motivation and Context

Resolves BED-6514

There was an edge case in the date range component used for File Ingest and Finished Jobs filters where you could only fix an invalid date range (ex. start comes after end) by changing the date which first caused the validation error. This fixes that issue so that updating either input to a valid state can resolve any date validation error.

## How Has This Been Tested?

Manually tested as well as ensuring previous automated integration tests still pass.

## Screenshots:

image

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

## Summary by CodeRabbit

- **New Features**
  - Managed Date Picker now uses string date values, returns an explicit validity flag on change, supports optional from/to hints and a validationError prop.
  - Date Range inputs provide separate start/end handlers that normalize values to day boundaries and validate start ≤ end.
  - Filter dialog now resets/undoes changes when opened/closed.

- **Bug Fixes**
  - Improved input/blur validation: clearing or invalid input preserves text, triggers validation, keeps calendar synchronized, and shows a clear end-date error when range order is invalid.

### CODE REVIEW

File: packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.test.tsx [hunk 1]
```diff
@@ -213,13 +213,13 @@ describe('ManagedDatePicker - value', () => {
         expect(onDateChangeMock).toHaveBeenCalledWith(new Date('2025-01-01T00:00:00.000Z'));
     });
 
-    it('does not updates value after blur when bad date typed', async () => {
+    it('sets value to undefined after blur when bad date typed', async () => {
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.test.tsx]: Some behaviors where changed to address an edge case. Tests were updated accordingly.

File: packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.tsx [hunk 2]
```diff
@@ -99,83 +119,79 @@ export const ManagedDatePicker: FC<Props> = ({
...
+    };
+
+    // If input string is a date, call onDateChange with date
+    const validateDateString = () => {
+        setPlaceholder(hint);
+
+        if (inputDateString === '') {
+            onDateChange(undefined);
             return;
         }
```
[the remark this answers is not in the document]
Comment (TheNando) [reply packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.tsx]: Not needed. The error will have already been cleared when the user typed.

**Comment by specter-flq** [comment]
Still looking at code but saw some weird behavior when running the UI locally. See video below. But the two things are:

- Clear All button doesn't clear date filter when dialog open. You need to close and open dialog for filters to be cleared
- Date filters don't reset when you close and open the dialog with cancel or clicking on background overlay. 

credit to @LucasParraF for suggesting I click that to test all functionality during a PR Review shadow session.

https://github.com/user-attachments/assets/88d49dc9-ba99-499b-8135-6bace4c33b1f

File: packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.test.tsx [hunk 3]
```diff
@@ -16,46 +16,32 @@
 
 import userEvent from '@testing-library/user-event';
 import { render, screen } from '../../test-utils';
-import { ManagedDatePicker, VALIDATIONS } from './ManagedDatePicker';
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.test.tsx:19]: Much of the validation logic was stripped out and simplified.

File: packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.tsx [hunk 4]
```diff
@@ -16,65 +16,30 @@
...
 import { DatePicker } from '@bloodhoundenterprise/doodleui';
 import { DateTime } from 'luxon';
-import { FC, useEffect, useId, useState } from 'react';
+import { FC, useEffect, useId, useRef, useState } from 'react';
 import { LuxonFormat, fiveYearsAgo, now } from '../../utils';
 
 const DATE_FORMAT_MASK = 'yyyy-mm-dd';
+const INVALID_DATE_MESSAGE = 'Input is not a valid date.';
 
-export const VALIDATIONS = {
```
Comment (TheNando) [inline packages/javascript/bh-shared-ui/src/components/ManagedDatePicker/ManagedDatePicker.tsx:24]: I kept running into edge cases so I took a different approach. The ManagedDatePicker now works with string values instead of Dates. It will only validate that the input is a valid date when the user types in the input. The validation to check end and start bounds is now handled in the DateRangeInput.

This is pretty much a full rewrite. It might be easier to just view the whole file rather than the changes.
